atom/common/api/event_emitter_caller.cc

```diff
@@ -20,8 +20,11 @@ v8::Local<v8::Value> CallMethodWithArgs(v8::Isolate* isolate,
                                    v8::MicrotasksScope::kRunMicrotasks);
   // Use node::MakeCallback to call the callback, and it will also run pending
   // tasks in Node.js.
-  return node::MakeCallback(isolate, obj, method, args->size(), &args->front(),
-                            {0, 0}).ToLocalChecked();
+  auto callback = node::MakeCallback(isolate, obj, method, args->size(), &args->front(),
+                            {0, 0});
+  if (callback.IsEmpty())
+    return v8::Local<v8::Object>();
+  return callback.ToLocalChecked();
```

Comment:
This is kind of strange, we're saying 
* if the callback is empty, return a v8::Local<v8::Object>
* otherwise, if the callback is empty, crash
* otherwise, return the value

These first two conditions are in conflict, so if we really want to remove the crash case here, better to replace this with a clearer one-liner:

`return callback.FromMaybe(v8::Local<v8::Object>{})`

I guess the question I'd ask though is why is node::MakeCallback failing? Is removing the crash here treating the symptom?

Reply:
@zcbenz is this potentially related? https://github.com/nodejs/nan/issues/284

If nothing else it seems like the behavior of MakeCallback has changed in the last few years?

Reply:
Ah after reading some more I think MakeCallback is probably fine. I'd like to understand why the callback ends up being empty. Is there no `emit` function on the JS side?

Reply:
@mgc I can see one code path inside node that would cause `MakeCallback` to be empty but I can't see what would trigger that path.

@ckerr Returning `v8::Object` was kinda just to see if it all worked, thinking it should either be `v8::Undefined` or just `v8::Value`.  I also don't understand your second condition, if the callback is empty the first condition (instant return) will occur the crash won't happen.